### PR TITLE
GH-61-1: User defined serializer and deserializer injection

### PR DESCRIPTION
**Pull request by kvmkreddy**

### CODE REVIEW

File: spring-kafka/src/test/java/org/springframework/kafka/annotation/EnableKafkaIntegrationTests.java [hunk 1]
```diff
@@ -265,7 +272,8 @@ public MultiListenerBean multiListener() {
 
 		@Bean
 		public ProducerFactory<Integer, String> producerFactory() {
-			return new DefaultKafkaProducerFactory<>(producerConfigs());
+			Map<String, Object> configs = producerConfigs();
+			return new DefaultKafkaProducerFactory<>(configs);
```
Comment (artembilan) [inline spring-kafka/src/test/java/org/springframework/kafka/annotation/EnableKafkaIntegrationTests.java]: Technically I don't see difference with existing code and your change.
Just some unnecessary noise...
Will revert on merge.

File: spring-kafka/src/test/java/org/springframework/kafka/listener/ConcurrentMessageListenerContainerTests.java [hunk 2]
```diff
@@ -173,6 +174,48 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
 	}
 
 	@Test
+	public void testAutoCommitWithCompositeStrategy() throws Exception {
```
Comment (artembilan) [inline spring-kafka/src/test/java/org/springframework/kafka/listener/ConcurrentMessageListenerContainerTests.java]: We don't have such a `strategy` any more. Therefore the test case reflect some parallel irrelevant reality.
Will fix on merge.

File: spring-kafka/src/test/java/org/springframework/kafka/listener/ConcurrentMessageListenerContainerTests.java [hunk 3]
```diff
@@ -173,6 +174,48 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
...
+		logger.info("Start auto");
+		Map<String, Object> props = KafkaTestUtils.consumerProps("test1", "true", embeddedKafka);
+
+		DefaultKafkaConsumerFactory<Integer, String> cf =
+				new DefaultKafkaConsumerFactory<Integer, String>(props,
+						new org.apache.kafka.common.serialization.IntegerDeserializer(),
+						new org.apache.kafka.common.serialization.StringDeserializer());
```
Comment (artembilan) [inline spring-kafka/src/test/java/org/springframework/kafka/listener/ConcurrentMessageListenerContainerTests.java]: M-m-m. I think we don't need such a fully qualified class names.
Just `import` should be enough.
Will fix on merge.

File: src/checkstyle/checkstyle.xml [hunk 4]
```diff
@@ -178,4 +178,4 @@
 		<module name="WhitespaceAround" />
 
 	</module>
-</module>
+</module>
```
Comment (artembilan) [inline src/checkstyle/checkstyle.xml]: We are coming back to this issue.
Each file has to have a new line in the end.
The IDE has a particular setting on the matter.

Although it is fully unclear what do you do in these files if we don't see any changes, only this noise :smile:

File: spring-kafka/src/main/java/org/springframework/kafka/core/DefaultKafkaConsumerFactory.java [hunk 5]
```diff
@@ -22,28 +23,55 @@
...
 	public DefaultKafkaConsumerFactory(Map<String, Object> configs) {
 		this.configs = new HashMap<>(configs);
 	}
 
+	public DefaultKafkaConsumerFactory(Map<String, Object> configs,
+			Deserializer<K> keyDeserializer,
+			Deserializer<V> valueDeserializer) {
+		this.configs = new HashMap<>(configs);
+		this.keyDeserializer = keyDeserializer;
+		this.valueDeserializer = valueDeserializer;
```
Comment (artembilan) [inline spring-kafka/src/main/java/org/springframework/kafka/core/DefaultKafkaConsumerFactory.java]: I understand that the `return new DefaultKafkaConsumerFactory(...)` looks good, but logically it is redundant extra noisy code.
But with respect to your contribution I leave it here and merge.
:crossed_flags:

File: src/checkstyle/checkstyle.xml (hunk 4 above)
Comment (kvmkreddy) [inline src/checkstyle/checkstyle.xml]: It seems some issues with my IDE which is corrupting this file now and then.

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-kafka/commit/48845e1de043ec9caf6493c7e2320966e7986a9f.

Thank you for the contribution!

Looking forward for more.

Hope the new PR won't be so tough on merge. :smile: 
The rebase and conflict fixing are really the branch author responsibility.

**Pull request closed without merging**